### ISSUE TITLE
Vs Code doesn't respect window priority after latest update.

### ISSUE THREAD

**Issue by joelmagner** [microsoft/vscode#126013]
Does this issue occur when all extensions are disabled?: Yes/No
Yes

Version: 1.57.0
Commit: b4c1bd0a9b03c749ea011b06c6d2676c8091a70c
Date: 2021-06-09T17:22:31.215Z (1 day ago)
Electron: 12.0.9
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: MacOS Catalina 10.15.7 x64 Intel

Steps to Reproduce:

1. Have Vs Code Window in focus
2. Try to tab (cmd+tab) to a different application. Vs code will always stay on top of the other windows. Tabbing from another application -> vs code works. But I cannot tab away from vs code.
I've tried restarting everything & killed all processes. Issue started with this latest release.

**Comment by bpasero** [comment]
This is very likely https://github.com/microsoft/vscode/issues/125122

**Comment by joelmagner** [comment]
@bpasero Don't think so. What I mean by Fullscreen is NOT equivalent to clicking the green fullscreen button. 
i.e. i can still see the status bar at top with the clock, battery, wifi etc.
TL;DR - size of window doesn't matter. It just wont go away when I tab. Will update issue.

**Comment by bpasero** [comment]
Hm yeah, it looks like no fullscreen is involved in your case. What are the exact steps for you?

**Comment by vu-dinh-hung** [comment]
I have the same problem. Cmd + Tabbing away from VS Code simply does not show the other applications. Downgraded VS Code and everything works again. This issue remain when I use another app for Cmd + Tab: [AltTab](https://alt-tab-macos.netlify.app/)

**Comment by bpasero** [comment]
Does this reproduce with just one window opening?

**Comment by deepak1556** [comment]
Can you attach a screen recording that describes the issue, be as detailed as possible with the steps.

**Comment by LukasKnuth** [comment]
I can reproduce with just a single VS Code window open.

Curiously, if I alt+tab _fast_ so that the application switcher overlay isn't even shown, it works as expected.

If I alt+tab, wait until the switcher overlay shows and then let go of alt, the desired window flashes very shortly and then VS Code is back on top.

**Comment by RoadToDream** [comment]
This is a minor but very annoyance bug, really hope this could be fixed. 
How to reproduce:
1. with a vs code window open.
2. switch to another window using cmd+tab, let switcher overlay appear.
3. vs code will stay on top and disregard the switch action.

**Comment by alk-sdavid** [comment]
Hello @deepak1556, you'll find a screen recording about this issue with alt-tab app here https://github.com/lwouis/alt-tab-macos/issues/992#issue-919175358

**Comment by cpnvr** [comment]
If it's helpful, here is the video link directly: https://www.loom.com/share/27bc61d2bcd94d3b8a49f78e82edfe62

**Comment by Neylion** [comment]
This heavily impacts my workflow in a negative way. Hope you can get a fix out for this soon! 🙏

**Comment by joelmagner** [comment]
> Can you attach a screen recording that describes the issue, be as detailed as possible with the steps.

Example of what information I can provide? If I create a screencapture all you're gonna see is me alt tabbing and nothing happening.

**Comment by alk-sdavid** [comment]
Issue fixed in Vscode v1.57.1 👍

**Comment by joelmagner** [comment]
> Issue fixed in Vscode v1.57.1 👍

@alk-sdavid For me it stopped working after a while again... 🤔

**Comment by karlhorky** [comment]
I have a possibly related issue, where the incorrect window is focused when multiple editor windows are open: https://github.com/microsoft/vscode/issues/126779

**Comment by jackdelab** [comment]
I am still having this same issue with VSCode 1.58.2
